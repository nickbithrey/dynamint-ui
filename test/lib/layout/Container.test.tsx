import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import {
    Container,
    IContainer
} from 'lib/layout/Container';

describe('<Container />', () => {
    
    const props: IContainer = {
        tokens: {
            childrenGap: 2,
            maxHeight: 20
        },
        styles: {
            inner: {}
        }
    }

    it('will render valid tag and matches snapshot', () => {
        const wrapper: ShallowWrapper = shallow(
            <Container {...props}>
                <p>test</p>
            </Container>
        );
        expect(wrapper).toMatchSnapshot();
    });
    
    it('will render null when no children are passed in', () => {
        const wrapper: ShallowWrapper = shallow(
            <Container {...props} />
        );
        expect(wrapper.html()).toBeNull();
    });
    
});