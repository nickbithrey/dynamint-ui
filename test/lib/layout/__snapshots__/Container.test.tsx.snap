// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> will render valid tag and matches snapshot 1`] = `
<Stack
  styles={
    Object {
      "inner": Object {},
    }
  }
  tokens={
    Object {
      "childrenGap": 2,
      "maxHeight": 20,
    }
  }
>
  <p>
    test
  </p>
</Stack>
`;
